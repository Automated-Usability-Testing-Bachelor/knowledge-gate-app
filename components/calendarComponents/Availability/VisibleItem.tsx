import React, { useCallback } from 'react'
import {
  View,
  Animated,
  TouchableHighlight,
  Text,
  StyleSheet
} from 'react-native'
import Colors from '../../../constants/Colors'
import { Availability } from '../../../mocks/CalendarMockData'
import ClockIcon from './clockIcon'

const styles = StyleSheet.create({
  rowFront: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    height: 60,
    margin: 5
  },
  rowFrontVisible: {
    backgroundColor: '#FFF',
    borderRadius: 5,
    height: 60,
    padding: 10
  },
  text: {
    fontSize: 14,
    lineHeight: 22,
    color: Colors.black.color
  }
})

export type Props = {
  onEditCallback: (id: string | number[]) => void
  data: Data
  rowHeightAnimatedValue: Animated.Value
  removeRow: () => void
  // eslint-disable-next-line react/no-unused-prop-types, @typescript-eslint/no-explicit-any
  leftActionState: any
  // eslint-disable-next-line react/no-unused-prop-types, @typescript-eslint/no-explicit-any
  rightActionState: any
}

export type Data = {
  index: number
  item: Availability
}

const VisibleItem: React.FC<Props> = (props) => {
  const {
    onEditCallback,
    data,
    rowHeightAnimatedValue,
    removeRow,
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    rightActionState
  } = props

  if (rightActionState) {
    Animated.timing(rowHeightAnimatedValue, {
      toValue: 0,
      duration: 200,
      useNativeDriver: false
    }).start(() => {
      removeRow()
    })
  }

  const onPress = useCallback(
    () => onEditCallback(data.item.id),
    [data.item.id, onEditCallback]
  )

  return (
    <Animated.View
      style={[
        styles.rowFront,
        {
          height: rowHeightAnimatedValue,
          justifyContent: 'center'
        }
      ]}
    >
      <TouchableHighlight
        style={[styles.rowFrontVisible, { justifyContent: 'center' }]}
        onPress={onPress}
        underlayColor={'#aaa'}
      >
        <View
          style={{
            justifyContent: 'center',
            flex: 1
          }}
        >
          <View style={{ flexDirection: 'row', alignItems: 'center' }}>
            <ClockIcon />
            <View>
              <Text style={styles.text}>{`${data.item.from.format(
                'hh:mm A'
              )} - ${data.item.to.format('hh:mm A')}`}</Text>
            </View>
          </View>
        </View>
      </TouchableHighlight>
    </Animated.View>
  )
}

export default VisibleItem
