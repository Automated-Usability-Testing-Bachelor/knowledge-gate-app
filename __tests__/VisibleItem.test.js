/*eslint-disable */

import React from 'react'
import VisibleItem from "../components/calendarComponents/Availability/VisibleItem"
import { fireEvent, render, waitFor, screen } from '@testing-library/react-native';
import {
  Animated,
} from 'react-native'
import moment from "moment"

// eslint-disable-next-line no-undef
describe('<VisibleItem />', () => {
    const id = "abc"
    const value = new Animated.Value(60)
    const dummyData = {index: 1, item: {id: id, from: moment(), to: moment()} }
    const onEditPress = jest.fn((id)=> id);
    it('Test onPress event', async () => {
        
        const activeButton = render(
        <VisibleItem onEditCallback={onEditPress} testID="button" rowHeightAnimatedValue={value} data={dummyData} />
        )
        const { getByTestId } = activeButton;
        const button = getByTestId('button');
        expect(activeButton).toBeDefined();

        const activeButtonAsJson = activeButton.toJSON();
        expect(activeButtonAsJson).toMatchSnapshot();
        fireEvent.press(button)
        fireEvent.press(button)
        await waitFor(() => {
            expect(onEditPress.mock.calls.length).toBe(2)
            expect(onEditPress.mock.results[0].value).toBe(id)
            expect(onEditPress.mock.results[1].value).toBe(id)
        });
      
    });
    it('Test text time from and to set', async () => {
        
        const activeButton = render(
        <VisibleItem onEditCallback={onEditPress} testID="button" rowHeightAnimatedValue={value} data={dummyData} />
        )
        const { getByText } = activeButton;
        const TextComp = getByText(`${dummyData.item.from.format(
                'hh:mm A'
              )} - ${dummyData.item.to.format('hh:mm A')}`);
        expect(TextComp).toBeDefined()
    });
  
});

