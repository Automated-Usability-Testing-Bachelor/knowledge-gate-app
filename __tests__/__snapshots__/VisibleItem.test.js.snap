// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VisibleItem /> Test onPress event 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#FFF",
      "borderRadius": 5,
      "height": 60,
      "justifyContent": "center",
      "margin": 5,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "backgroundColor": "#FFF",
          "borderRadius": 5,
          "height": 60,
          "padding": 10,
        },
        {
          "justifyContent": "center",
        },
      ]
    }
    testID="button"
  >
    <View
      style={
        {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "paddingRight": 5,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            aria-hidden="true"
            bbHeight={24}
            bbWidth={24}
            color={4288652970}
            fill="none"
            focusable={false}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="currentColor"
            strokeWidth={1.5}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            tintColor={4288652970}
            vbHeight={24}
            vbWidth={24}
            width={24}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              stroke={
                [
                  2,
                ]
              }
              strokeWidth={1.5}
            >
              <RNSVGPath
                d="M12 6v6h4.5m4.5 0a9 9 0 11-18 0 9 9 0 0118 0z"
                propList={
                  [
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                strokeLinecap={1}
                strokeLinejoin={1}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View>
          <Text
            style={
              {
                "color": "#1D1D1B",
                "fontSize": 14,
                "lineHeight": 22,
              }
            }
          >
            04:48 PM - 04:48 PM
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
